ytt: document openapi validation attributes, mapped to named validations

peschmae opened the pull request:
Documentation for [ytt#899](https://github.com/carvel-dev/ytt/pull/899)

peschmae committed:
ytt: document openapi validation attributes, mapped to named validations

Signed-off-by: Mathias Petermann <mathias.petermann@gmail.com>

joaopapereira commented on site/content/ytt/docs/develop/how-to-export-schema.md:
```diff
@@ -221,10 +230,56 @@ The value of this property:
 
+### `minimum`
+
+(As of vT.B.D+)
```
The documentation is versioned so we do not need this
```suggestion
```

joaopapereira commented on site/content/ytt/docs/develop/how-to-export-schema.md:
```diff
@@ -221,10 +230,56 @@ The value of this property:
+
+### `maximum`
+
+(As of vT.B.D+)
```
The documentation is versioned so we do not need this
```suggestion
```

joaopapereira commented on site/content/ytt/docs/develop/how-to-export-schema.md:
```diff
@@ -221,10 +230,56 @@ The value of this property:
+
+### `minLength`
+
+(As of vT.B.D+)
```
The documentation is versioned so we do not need this
```suggestion
```

joaopapereira commented on site/content/ytt/docs/develop/how-to-export-schema.md:
```diff
@@ -221,10 +230,56 @@ The value of this property:
+
+### `maxLength`
+
+(As of vT.B.D+)
```
The documentation is versioned so we do not need this
```suggestion
```

joaopapereira commented on site/content/ytt/docs/develop/how-to-export-schema.md:
```diff
@@ -221,10 +230,56 @@ The value of this property:
+
+### `enum`
+
+(As of vT.B.D+)
```
The documentation is versioned so we do not need this
```suggestion
```

joaopapereira reviewed (changes requested):
Thanks for the PR, added some comments, but generally looks good

peschmae committed:
Remove version hint for named validations

Signed-off-by: Mathias Petermann <mathias.petermann@gmail.com>

peschmae committed:
ytt: add hint about when= supressing validation export, add minItems/minProperties mapping for array and objects

Signed-off-by: Mathias Petermann <mathias.petermann@gmail.com>

peschmae commented:
Updated documentation with a hint that validations aren't exported when a `when=` condition is set on a validation.

Added `minItems`/`minProperties` attribute documentation for array and objects added in https://github.com/carvel-dev/ytt/commit/43dee5fb14f62a344c600252fc1980bda5800e76

joaopapereira reviewed (approved):
LGTM
